### DISCUSSION TITLE
Add Files with Framework

### DISCUSSION THREAD

**Discussion by WebPirat** [pocketbase/pocketbase#5020] [Q&A] [answered]
Hello,

I didn't find it in the documentation. Is there an option to upload files directly in Go?

Best Regards

**Comment by ganigeorgiev** [comment, marked as answer]
At the moment record files can be added programmatically when using the [forms.RecordUpsert](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.22.13/forms#RecordUpsert):

```go
form := forms.NewRecordUpsert(app, record)

// manually upload file(s)
// (for other factories check https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.22.13/tools/filesystem#File)
f1, _ := filesystem.NewFileFromPath("/path/to/file1")
f2, _ := filesystem.NewFileFromPath("/path/to/file2")
form.AddFiles("yourFileField1", f1, f2)

err := form.Submit()
```
(_see also https://pocketbase.io/docs/go-records/#create-new-record-with-data-validations_)

Note that this will be simplified in the near future as part of the [ongoing refactoring](https://github.com/pocketbase/pocketbase/discussions/4355).

**Reply by WebPirat** [reply]
oh men overlooked that sorry thanks for your help and time
